Extension/src/LanguageServer/client.ts

```diff
@@ -1067,6 +1133,9 @@ export class DefaultClient implements Client {
                             if (settings.codeFolding) {
                                 this.codeFoldingProviderDisposable = vscode.languages.registerFoldingRangeProvider(this.documentSelector, new FoldingRangeProvider(this));
                             }
+                            if (settings.enhancedColorization && this.semanticTokensLegend) {
```

Comment:
Why is this check needed? Isn't it alway strue?

Reply:
I assume you're referring to `this.semanticTokensLegend`, as the setting is not always true.  This check was needed because the variable is nullable.  It should always be true here, but the check is necessary to silence the lint error.